### ISSUE TITLE
Images embedded in PDF disappearing

### ISSUE THREAD

**Issue by whifur** [mozilla/pdf.js#11328]
I am using this test link [https://mozilla.github.io/pdf.js/web/viewer.html](https://mozilla.github.io/pdf.js/web/viewer.html)

The test pdf [pdfjs_lost_image_test.pdf](https://github.com/mozilla/pdf.js/files/3840182/pdfjs_lost_image_test.pdf)

![pdfjs_2](https://user-images.githubusercontent.com/17464719/68747444-84d7f400-0635-11ea-8b36-d4902b4a359b.png)

The image is lost when previewing with pdfjs.

![pdfjs_1](https://user-images.githubusercontent.com/17464719/68747462-8d302f00-0635-11ea-8d76-347e00081066.png)

A temporary solution is to convert the file itself, convert this pdf to jpg first, and then convert this jpg to pdf.

Is it possible that the font is Chinese traditional? and how can I solve it?

Thanks a million!

**Comment by timvandermeij** [comment]
I don't see the images either using Firefox 70.0.1 (64-bit, Linux). Thank you for reporting this.

**Comment by mepowerlmay** [comment]
I have some question.....could somebody help this?

**Comment by cetinsert** [comment]
There is a regression in `SVG` rendering of the page in `2.10.377` ❌. (latest known working version `2.8.335` ✔️).

```
Error: Unknown IR type: p
    at i._makeShadingPattern
    at i.shadingFill
    at i.executeOpTree
    at i.group
    at i.executeOpTree
```

**Comment by Snuffleupagus** [comment]
> There is a regression in `SVG` rendering of the page

Please note that as mentioned in the Wiki, see https://github.com/mozilla/pdf.js/wiki/Frequently-Asked-Questions#backends, the SVG-backend is *not officially supported*.
Furthermore, this issue is about the default (i.e. `canvas`) backend specifically; hence let's avoid confusing things by not trying to track multiple things here :-)

**Comment by cetinsert** [comment]
> *not officially supported*

means do not bother us with reports, we are privately messing with it? I am good with that :-)

I just truly love that backend and wish all the best for it!

**Comment by Snuffleupagus** [comment]
This is now getting really off-topic, but as explained previously the SVG-backend doesn't perform well enough for most real world applications (since it leads to *huge* number of DOM elements which browsers don't handle well); hence why development stopped.

**Comment by Snuffleupagus** [comment]
This *mostly* fixed by PR #14175 (only the cloud with the "Ethernet" text looks wrong now).
